store/tikv/2pc.go

```diff
@@ -526,6 +535,13 @@ func (actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoffer, bat
 		prewriteResp := resp.Resp.(*pb.PrewriteResponse)
 		keyErrs := prewriteResp.GetErrors()
 		if len(keyErrs) == 0 {
+			if bytes.Equal(c.primary(), batch.keys[0]) {
+				// After writing the primary key, if the size of the transaction is large than 4M,
+				// start the ttlManager. The ttlManager will be closed in tikvTxn.Commit().
+				if c.txnSize > 4*1024*1024 {
+					c.run(c, nil)
```

Comment:
The ttlManager use `PessimisticLockTTL`, it assumes the primary key is pessimistic.

Reply:
Need to pass `kill` pointer?

Reply:
It's not easy to pass that variable in the current code.
I'll tiny refactor the code later.